modules/appnexusBidAdapter.js

```diff
@@ -961,6 +961,7 @@ function bidToTag(bid) {
     tag['banner_frameworks'] = bid.params.frameworks;
   }
 
+  // TODO: why does this need to iterate through every ad unit?
```

Comment:
I think the idea was to look at the adUnit as the source of truth here, since PBJS (at the time at least, not sure if it's still true today) could create a backup mediaTypes object in the bid object if the mediaTypes field was undefined in the setup.  In that scenario, we didn't want to pass along the ad_type field if the publisher didn't define it in their adUnit.  (see https://github.com/prebid/Prebid.js/pull/4065 for the original PR from ~4 years ago).

There's probably other ways this can be done nowadays, so feel free to give it a try if it will help clean things up for the rest of the project.

Reply:
I don't understand the reasoning. If prebid says that the default mediatype is banner, why would appnexus want to contradict that? and why would _only_ appenxus want to contradict that? I'd expect all bidders to give me a banner if that's what I am (implicitly) asking for.

Besides that, there are some other scenarios where this would behave surprisingly I think.  it's possible to set up a multiformat adUnit and use sizemapping to turn off some mediatypes for some form factors;  for example banner & video but banner is disabled on mobile. Granted it's not very likely, but this logic would circumvent it.

Reply:
seems like a new issue would work? this pr is merging and unrelated, if i'm reading this correctly